File: python/examples/calculator.py
"""
Ví dụ: Máy tính đơn giản
"""

def calculator():
    """Máy tính đơn giản với các phép toán cơ bản"""
    
    print("=== Máy tính đơn giản ===")
    print("1. Cộng")
    print("2. Trừ")
    print("3. Nhân")
    print("4. Chia")
    print("5. Thoát")
    
    while True:
        choice = input("\nChọn phép toán (1-5): ")
        
        if choice == '5':
            print("Tạm biệt!")
            break
        
        if choice in ['1', '2', '3', '4']:
            try:
                num1 = float(input("Nhập số thứ nhất: "))
                num2 = float(input("Nhập số thứ hai: "))
                
                if choice == '1':
                    result = num1 + num2
                    print(f"Kết quả: {num1} + {num2} = {result}")
                elif choice == '2':
                    result = num1 - num2
                    print(f"Kết quả: {num1} - {num2} = {result}")
                elif choice == '3':
                    result = num1 * num2
                    print(f"Kết quả: {num1} * {num2} = {result}")
                elif choice == '4':
                    if num2 == 0:
                        print("Lỗi: Không thể chia cho 0!")
                    else:
                        result = num1 / num2
                        print(f"Kết quả: {num1} / {num2} = {result}")
            except ValueError:
                print("Lỗi: Vui lòng nhập số hợp lệ!")
        else:
            print("Lựa chọn không hợp lệ!")

if __name__ == "__main__":
    calculator()

File: rust/examples/calculator.rs
use std::io;

fn main() {
    loop {
        println!("\n=== Máy tính đơn giản ===");
        println!("1. Cộng");
        println!("2. Trừ");
        println!("3. Nhân");
        println!("4. Chia");
        println!("5. Thoát");
        
        print!("\nChọn phép toán (1-5): ");
        io::Write::flush(&mut io::stdout()).unwrap();
        
        let mut choice = String::new();
        io::stdin()
            .read_line(&mut choice)
            .expect("Failed to read line");
        
        let choice = choice.trim();
        
        if choice == "5" {
            println!("Tạm biệt!");
            break;
        }
        
        if choice == "1" || choice == "2" || choice == "3" || choice == "4" {
            print!("Nhập số thứ nhất: ");
            io::Write::flush(&mut io::stdout()).unwrap();
            let mut num1_str = String::new();
            io::stdin()
                .read_line(&mut num1_str)
                .expect("Failed to read line");
            
            print!("Nhập số thứ hai: ");
            io::Write::flush(&mut io::stdout()).unwrap();
            let mut num2_str = String::new();
            io::stdin()
                .read_line(&mut num2_str)
                .expect("Failed to read line");
            
            let num1: f64 = match num1_str.trim().parse() {
                Ok(num) => num,
                Err(_) => {
                    println!("Lỗi: Vui lòng nhập số hợp lệ!");
                    continue;
                }
            };
            
            let num2: f64 = match num2_str.trim().parse() {
                Ok(num) => num,
                Err(_) => {
                    println!("Lỗi: Vui lòng nhập số hợp lệ!");
                    continue;
                }
            };
            
            match choice {
                "1" => {
                    let result = num1 + num2;
                    println!("Kết quả: {} + {} = {}", num1, num2, result);
                }
                "2" => {
                    let result = num1 - num2;
                    println!("Kết quả: {} - {} = {}", num1, num2, result);
                }
                "3" => {
                    let result = num1 * num2;
                    println!("Kết quả: {} * {} = {}", num1, num2, result);
                }
                "4" => {
                    if num2 == 0.0 {
                        println!("Lỗi: Không thể chia cho 0!");
                    } else {
                        let result = num1 / num2;
                        println!("Kết quả: {} / {} = {:.2}", num1, num2, result);
                    }
                }
                _ => {}
            }
        } else {
            println!("Lựa chọn không hợp lệ!");
        }
    }
}

